src/test/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogViewModelTest.java

```diff
@@ -0,0 +1,126 @@
+                taskExecutor
+        );
+    }
+
+    @Test
+    public void testStartImportWithValidFiles() throws Exception {
+        // Create temporary test files
+        Path tempDir = Files.createTempDirectory("testDir");
+        Path subDir = tempDir.resolve("subdir");
+        Files.createDirectories(subDir);
+
+        // Create test files: one in the main directory and one in the subdirectory
+        Path tempFile1 = Files.createTempFile(tempDir, "file1", ".pdf"); // Main directory
+        Path tempFile2 = Files.createTempFile(subDir, "file2", ".txt");
+
+        // Arrange: Mock file nodes with the absolute paths of the temporary files
+        FileNodeViewModel fileNode1 = mock(FileNodeViewModel.class);
+        FileNodeViewModel fileNode2 = mock(FileNodeViewModel.class);
+
+        // Mock the getPath method to return the expected paths
+        when(fileNode1.getPath()).thenReturn(tempFile1);
+        when(fileNode2.getPath()).thenReturn(tempFile2);
+
+        // Create TreeItem for each FileNodeViewModel
+        TreeItem<FileNodeViewModel> treeItem1 = new TreeItem<>(fileNode1);
+        TreeItem<FileNodeViewModel> treeItem2 = new TreeItem<>(fileNode2);
+
+        // Initialize SimpleListProperty and assign ObservableList
+        SimpleListProperty<TreeItem<FileNodeViewModel>> checkedFileListProperty =
+                new SimpleListProperty<>(FXCollections.observableArrayList(treeItem1, treeItem2));
+
+        // Assert that the list contains 2 items
+        assertEquals(2, checkedFileListProperty.get().size());
+        assertEquals(tempFile1, checkedFileListProperty.get().get(0).getValue().getPath());
+        assertEquals(tempFile2, checkedFileListProperty.get().get(1).getValue().getPath());
+
+        Path directory = tempDir; // Base directory for relativization
+
+        // Create list of relative paths
+        List<Path> fileList = checkedFileListProperty.stream()
+                                                     .map(item -> item.getValue().getPath())
+                                                     .filter(path -> path.toFile().isFile())
+                                                     .map(path -> directory.toAbsolutePath().relativize(path.toAbsolutePath())) // Convert to relative path
+                                                     .collect(Collectors.toList());
+
+        // Check if fileList is empty
+        if (fileList.isEmpty()) {
+            LOGGER.warn("There are no valid files checked");
+            return;
+        }
+
+        assertEquals(2, fileList.size(), "fileList should contain 2 paths");
+        assertTrue(fileList.contains(directory.toAbsolutePath().relativize(tempFile1.toAbsolutePath())),
+                "fileList should contain the relative path of file1.pdf");
+        assertTrue(fileList.contains(directory.toAbsolutePath().relativize(tempFile2.toAbsolutePath())),
+                "fileList should contain the relative path of file2.txt");
```

Comment:
Do one `assertEquals`

    assertEquals(List.of(directory.toAbsolutePath().relativize(tempFile1.toAbsolutePath()), fileList.contains(directory.toAbsolutePath().relativize(tempFile2.toAbsolutePath())), fileList)


Then, you checked both `size` and the containment.